vector/src/main/java/fr/gouv/tchap/features/home/room/list/TchapRoomSummaryItem.kt

```diff
@@ -0,0 +1,203 @@
+        holder.rootView.setOnLongClickListener {
+            it.performHapticFeedback(HapticFeedbackConstants.LONG_PRESS)
+            itemLongClickListener?.onLongClick(it) ?: false
+        }
+        holder.titleView.text = TchapUtils.getNameFromDisplayName(matrixItem.getBestName())
+        holder.lastEventTimeView.text = lastEventTime
+        holder.lastEventView.text = lastFormattedEvent
+        holder.unreadCounterBadgeView.text = RoomSummaryFormatter.formatUnreadMessagesCounter(unreadNotificationCount)
+        holder.unreadIndentIndicator.isVisible = hasUnreadMessage
+        holder.draftView.isVisible = hasDraft
+        renderAvatar(holder)
+        renderRoomType(holder)
+        renderRoomAccessView(holder)
+        renderSelection(holder, showSelected)
+        holder.typingView.setTextOrHide(typingMessage)
+        holder.lastEventView.isInvisible = holder.typingView.isVisible
+        holder.unreadCounterBadgeView.manageVisibility(unreadNotificationCount > 0, true)
+        holder.disabledNotificationsBadge.manageVisibility(hasDisabledNotifications, true)
+        holder.expectedActionBadgeView.manageVisibility(hasExpectedAction, false)
+    }
+
+    override fun unbind(holder: Holder) {
+        holder.rootView.setOnClickListener(null)
+        holder.rootView.setOnLongClickListener(null)
+        avatarRenderer.clear(holder.avatarImageView)
+        super.unbind(holder)
+    }
+
+    private fun renderSelection(holder: Holder, isSelected: Boolean) {
+        if (isSelected) {
+            holder.avatarCheckedImageView.visibility = View.VISIBLE
+            val backgroundColor = ContextCompat.getColor(holder.view.context, R.color.riotx_accent)
+            val backgroundDrawable = TextDrawable.builder().buildRound("", backgroundColor)
+            holder.avatarImageView.setImageDrawable(backgroundDrawable)
+        } else {
+            holder.avatarCheckedImageView.visibility = View.GONE
+            avatarRenderer.render(matrixItem, holder.avatarImageView)
+        }
+    }
+
+    private fun renderAvatar(holder: Holder) {
+        holder.avatarImageView.visibility = if (isDirect) View.VISIBLE else View.GONE
+        holder.avatarHexagonImageView.visibility = if (isDirect) View.GONE else View.VISIBLE
+
+        avatarRenderer.render(
+                matrixItem,
+                if (isDirect)
+                    holder.avatarImageView
+                else holder.avatarHexagonImageView
+        )
+    }
+
+    private fun renderRoomType(holder: Holder) {
+        if (isDirect)
+            holder.domainNameView.text = TchapUtils.getDomainFromDisplayName(matrixItem.getBestName())
+        else {
+            /**
+             * FIXME : handle state of each room based on RoomAccessRules (from v1)
+             * By default, it will be "Private"
+             */
+            holder.domainNameView.apply {
+                val roomTypeString: Int
+                val roomTypeColor: Int
+
+                when (roomType) {
```

Comment:
Please consider here the case of RoomTchapType.Direct (instead of considering it separately just above) - For that you will have to use your method `getRoomType` to set the roomType (with the current available information)

Reply:
About the case of a direct chat, please set the text color. I observed wrong text color by testing the application